services/streammanager/googlesheets/googlesheetsmanager.go

```diff
@@ -50,34 +70,59 @@ func NewProducer(destinationConfig interface{}) (*sheets.Service, error) {
 	if err != nil {
 		return nil, fmt.Errorf("[GoogleSheets] error  :: error in GoogleSheets while unmarshalling destination config:: %w", err)
 	}
-	if config.Credentials != "" {
-		err = json.Unmarshal([]byte(config.Credentials), &credentialsFile)
-		if err != nil {
-			return nil, fmt.Errorf("[GoogleSheets] error  :: error in GoogleSheets while unmarshalling credentials json:: %w", err)
+
+	var opts []option.ClientOption = make([]option.ClientOption, 0)
+
+	if config.TestConfig.Endpoint != "" {
+		opts = append(opts, option.WithEndpoint(config.TestConfig.Endpoint))
+		token := &oauth2.Token{
+			AccessToken:  config.TestConfig.AccessToken,
+			RefreshToken: config.TestConfig.RefreshToken,
+		}
+		config := &tls.Config{
+			// skipcq: GSC-G402
+			InsecureSkipVerify: true,
 		}
+		client := oauth2.NewClient(context.Background(), oauth2.StaticTokenSource(token))
+		trans := client.Transport.(*oauth2.Transport)
+		trans.Base = &http.Transport{TLSClientConfig: config}
+		opts = append(opts, option.WithHTTPClient(client))
+	} else {
+		if config.Credentials != "" {
+			err = json.Unmarshal([]byte(config.Credentials), &credentialsFile)
+			if err != nil {
+				return nil, fmt.Errorf("[GoogleSheets] error  :: error in GoogleSheets while unmarshalling credentials json:: %w", err)
+			}
 
-	}
-	// Creating token URL from Credentials file if not using constant from google.JWTTOkenURL
-	tokenURI := google.JWTTokenURL
-	if len(credentialsFile.TokenUrl) != 0 {
-		tokenURI = credentialsFile.TokenUrl
-	}
-	// Creating JWT Config which we are using for getting the oauth token
-	jwtconfig := &jwt.Config{
-		Email:      credentialsFile.Email,
-		PrivateKey: []byte(credentialsFile.PrivateKey),
-		Scopes: []string{
-			"https://www.googleapis.com/auth/spreadsheets",
-		},
-		TokenURL: tokenURI,
+		}
```

Comment:
Is there a reason for nesting this inside the `else {` 

We only want to check for credentials if no Endpoint is provided?

Reply:
refactored